packages/gatsby/src/redux/run-sift.js

```diff
@@ -41,11 +39,11 @@ const getFilters = filters =>
 // Run Sift
 /////////////////////////////////////////////////////////////////////
 
-function isEqId(firstOnly, fieldsToSift, siftArgs) {
+function isEqId(firstOnly, siftArgs) {
   return (
     firstOnly &&
-    Object.keys(fieldsToSift).length === 1 &&
-    Object.keys(fieldsToSift)[0] === `id` &&
+    siftArgs.length > 0 &&
```

Comment:
should this be `siftArgs.length === 1`? if we have more than one filter, this special case does not apply?

Reply:
Yes, thanks!